crates/bevy_pbr/src/render/shadows.wgsl

```diff
@@ -141,25 +137,9 @@ fn sample_cascade(light_id: u32, cascade_index: u32, frag_position: vec4<f32>, s
     let light_local = offset_position_ndc.xy * flip_correction + vec2<f32>(0.5, 0.5);
 
     let depth = offset_position_ndc.z;
-    // do the lookup, using HW PCF and comparison
-    // NOTE: Due to non-uniform control flow above, we must use the level variant of the texture
-    // sampler to avoid use of implicit derivatives causing possible undefined behavior.
-#ifdef NO_ARRAY_TEXTURES_SUPPORT
-    return textureSampleCompareLevel(
```

Comment:
Why did this use textureSampleCompareLevel() here, in the original code before this PR? The  NO_ARRAY_TEXTURES_SUPPORT for point shadows used just textureSampleCompare(). Switching it to the latter in my sample_shadow_map_hardware() function fixed the WebGL2 crashes.

Reply:
See my other comment on this in the place where the code was moved. I want to double-check with cwfitzgerald. :)

Reply:
As mentioned, bother blocks are needed and both should be textureSampleCompareLevel.